### ISSUE TITLE
welcome

### ISSUE THREAD

**Issue by derfelot** [derfelot/discussion#1]
This can be used as a place for discussion of cm12.1 security updates

ping @MSe1969 @Micha-Btz

**Comment by Micha-Btz** [comment]
ok, will we discuss in english or german?

how we will begin? did we merge my patches first or will martin upload patches for April and May?
And i will suggest a list of CVS that we didn't fixes maybe someone later there is a new possibility.

**Comment by derfelot** [comment]
I wouldn't mind either German or English. but in case somebody else would like to join later it might be a good idea to keep it in English.

I agree, we should make a list of all the April/may patches. then you guys can merge your stuff (maybe on a test branch first) which we can then cross off the list and we'll see what's left.

**Comment by MSe1969** [comment]
Hi,

quick update:

1.] 
I've added the file giving the overview about what I was able to patch and what not. There is in fact 2 changes belonging to the April 1st list, where - due to significant code changes - a patch is not trivial (at least not for me :) ).
From the April 5th list, there are six kernel changes, where it would need to be determined, whether the changes are even relevant and if yes, how to backport them properly.
Maybe you can have a look and either take up some tasks or help deciding, which of the kernel changes in question are maybe truly irrelevant.

2.]
As far as I can see, only the repo owner is able to create a new branch, so @derfelot , could you please create a cm-12.1-test branch for the following repos or, if needed, create/fork the repo first:

build
external/libhevc
external/libnl
external/skia
external/sonivox
frameworks/av
frameworks/base
frameworks/native
packages/services/Telephony
kernel/sony/msm8974

3.]
I have in my own environment also a patch file for vendor/cm to simply exclude 'CMUpdater' from being built and included. (And a further patch in build, which I'll discuss in a PM on XDA)
There are different opinions about which cm parts are really necessary and which not and numerous scripts exist which help removing stuff during the flash process, so in general, I tend to leave anything in, so people should decide themselves, what to remove. However - CMUpdater will under no circumstances work: CM site is down forever and LOS will definitely never provide CM12.1 updates for our ROM. So I suggest to fork vendor/cm and start 'tweaking' it as well. (However, that is low priority)

Regards, M.

**Comment by derfelot** [comment]
Hi,

thanks for the update.

1) Thanks for the list. I will have a look at those not so trivial patches and see what i can come up with.

2) All the repos you mentioned are now present and have a cm-12.1-test branch. Merge away :)

I also included the local_manifest, so we can change the cm-12.1-test branch there to use the other cm-12.1-test branches to try out builds.

3) I agree in leaving in as much as possible, as it should be up to other people what they want / need and just remove the rest. I do agree that the CMUpdater app is no longer useful at all, but without it the settings crash when being selected (IIRC). Do you also have a patch to remove it from there completely?

Cheers

**Comment by MSe1969** [comment]
Hi derf,

thanks a lot - I'll later start with my work.
Regarding the CMUpdater remove, works without any crash - I'll put my patch file into this repo.

Last question rgd. commits:
Do you have any preference for the initial commit? (e.g. one commit per CVE, even if that means several commits in the same repo; one commit per repo, or even one parent commit for all repo's)  
I would normally commit per repo, as my patch files currently are structured per repo.

Regards, M.

**Comment by MSe1969** [comment]
Hi, 'android_external_libhevc' has no 'cm-12.1-test' branch yet.
Could you pls. add there as well? Thanks

**Comment by derfelot** [comment]
sorry, must have slipped through the cracks, it's there now.

as for commits, I think over commit per cve is would make it more "readable".

**Comment by MSe1969** [comment]
Thanks, 
as you can see, I have started committing my changes into the respective 'cm-12.1-test' branches, but I am not finished, yet, will continue tomorrow.
Once I am fully done, I'll do a test build and then let you know.
Regards, M.

**Comment by Micha-Btz** [comment]
my patches for April misses CVE-2017-0558 and 0560
@MSe1969 did you have become this merged?

**Comment by Micha-Btz** [comment]
should i renew my PR to the test branches or @MSe1969 can you merge all patches for april and may?

we can also remove some useless stuff. personally i use https://github.com/Micha-Btz/freecyngn
for this because I had have trouble to build the rom with such a mass removes.

**Comment by MSe1969** [comment]
Hi @Micha-Btz ,

this discussion repo has got some files in the code section. I have listed all patches in the file 'April-Patches-Overview.txt' with a status whether implemented, not relevant or me not having been able to backport: I was, same as you, so far also not able to work on CVE-2017-0558 and 0560 either.

The two of us can now directly commit, so no PR necessary. I will later this morning finish my commits into the test branches. However, that should not hinder you from helping - we should only make sure not to work at the same time on the same repo. If you tell me, which repo's you would like to take care now, I will skip those - please note, derf and I have agreed to make commit messages per CVE.
Currently still outstanding from my side are: frameworks/av, frameworks/base, packages/services/Telephony and kernel/sony/msm8974

As for removing useless stuff - we had discussed yesterday not to remove any stuff, unless it is definitely non-functional, like e.g. CMUpdater. @derfelot could you please fork vendor/cm - so we can exclude CMUpdater from the build? (see the patch file in this repo, doing it this way is clean, it is then simply not part of the settings any longer and not put into the build - no side effects)
Everybody using the ROM later is of course free to use e.g. your script to get rid of further stuff.

I suggest when we have done all April stuff, a test build should be made and we should do some testing and then publish the April build, before moving on to May.
I've encountered for my osprey device a funny situation for May (I keep 'cm-13.0' alive for this one): All critical CVE's, except one, are trivial to patch, but after patching, it does not work properly ... :(

**Comment by Micha-Btz** [comment]
ok, then i would commit my PR to the repo so you can skip 
framework native/av and telephony - on the kernel side i didn't have success with cherry picking.
@derfelot how did you do this? did you have an extra repo from hammerhead and generate a patch from git or add you hammerhead as remote repo to the sony stuff?

I actually run a rom, that contains all the patches i have made to my repos and it run very well. but i agree to build a rom after commit the april patches and then go on to may.

**Comment by MSe1969** [comment]
Sounds good - btw, in fact all three frameworks not yet done by me (av, base, native), so I will skip them and Telephony and only do the kernel stuff now.
With regards to the kernel stuff, I did not even try to cherry-pick, but went through it manually, looked at the diff view, checked whether the files would exists, copy/pasted patch file out of diff view and tried patch --dry-run first to see how it looks like. That is why I have from the April 5th section six patches remaining, which are "non-trivial" and where I have no clue whether they are even relevant.

**Comment by derfelot** [comment]
I will do the fork later today.

as for the kernel patches, the manual way is fine. you can also manually copy/paste them from other devices. it's not how it's "supposed" to be done, but I've done it like that myself many times. in the end, this is nothing official and as long as it works and patches are applied, that's fine for me ;)

as for cherry-picking, you should be able to clone our kernel, add another one as remote branch, fetch all, cherry pick and push back

**Comment by MSe1969** [comment]
Okay, I have now completed:
build
external/libhevc
external/libnl
external/skia
external/sonivox
packages/services/Telephony
kernel/sony/msm8974

this means, the following is still outstanding and will be done by @Micha-Btz :
frameworks/av
frameworks/base
frameworks/native

And to summarize, the following CVE's of the April '17 ASB are not (yet) implemented:
CVE-2017-0558, moderate, frameworks/av
CVE-2017-0560,  moderate, frameworks/base
CVE-2016-10232,  High, kernel
CVE-2014-9922,  High, kernel
CVE-2016-5349,  High, kernel
CVE-2016-10234,  High, kernel
CVE-2016-7097,  moderate, kernel
CVE-2016-5346,  moderate, kernel

**Comment by derfelot** [comment]
great, thank you. vendor_cm is there now.

**Comment by MSe1969** [comment]
thanks as well, 
patched vendor/cm in 'cm-12.1-test' and updated also local_manifests accordingly

I suggest, once @Micha-Btz is done, we make a build, test and if all fine, merge cm-12.1-test into cm-12.1

**Comment by Micha-Btz** [comment]
I will merge the patches today evening.

**Comment by Micha-Btz** [comment]
so, patches merged. i had also include the fix for CVE-2017-0558.
After a detailed look at CVE-2017-0560 i have to give up. there are a hell lot of differences in code.

**Comment by Micha-Btz** [comment]
fix for CVE-2014-9922 is not needed our kernel doesn't support sdcardfs

**Comment by derfelot** [comment]
Great, I will see about those remaining ones

**Comment by MSe1969** [comment]
Thanks a lot, cool - I am fine to remove the three.

Regarding CVE-2017-0560, I personally don't care either: 
It's about remote wipe, triggered by an app being device admin (e.g. Google Spy Services). I don't fully get, what the code and the commit comments want to say, but somehow seems, that if a secondary 'android user' was setup and active (instead of the default primary user), there could have been some problems... - maybe worth a try to look into MM (cm-13) code? Besides, not really that relevant (and old Z1C is not the most attractive thing to steal and specialists who are keen on the phone's data are not really impressed at all by the Google remote wipe feature... :) )

**Comment by derfelot** [comment]
CVE-2016-10234 I already merged in January it seems, so it can go.

I also found a backport for CVE-2014-9922 that I will include anyway, even if we don't have sdcardfs.

CVE-2016-10232 doesn't concern us indeed, we don't even have the corresponding functions.

**Comment by MSe1969** [comment]
Have built now and flashed, so far all ok :)

**Comment by Micha-Btz** [comment]
I also have done a build. no problems so far.

**Comment by MSe1969** [comment]
Hi, 
for the May ASB, we need two new forks: 
external/bluetooth/bluedroid
external/giflib

I'll soon put a file with the May statuses into this repo.
Regards, M.

**Comment by MSe1969** [comment]
file with 1st overview about May patches now in discussion repo - please have a look.
I did not really deep-dive into the code, yet - where I could easily patch, I did already, where the code was different, I simply indicated, without really looking at the topic, yet.

**Comment by Micha-Btz** [comment]
do we further commit to the test branches or will you first merge the patches to the cm-12.1 branches?
for april is at least 
CVE-2017-0560, moderate, frameworks/base
CVE-2016-5349, High, kernel
open, i put in a fix for  CVE-2016-7097. I think CVE-2017-0560 can we ignore as @MSe1969 said.

**Comment by derfelot** [comment]
So is CVE-2016-5349 is already fixed then? You removed it from the list.

I will go ahead and merge everything into the cm-12.1 branches and open up test branches for the May stuff.

Ill open a new issue for the remaining April stuff (if needed) and for May to avoid too much cluttering in this one.

**Comment by derfelot** [comment]
OK, everything is merged into the cm-12.1 branches. I will go through may and create what is needed there.

**Comment by Micha-Btz** [comment]
merde,  CVE-2016-5349 is not fixed, fixed is   https://review.lineageos.org/#/c/166122/ 
CVE-2016-5346 which was already in. I will correct that. right is what i have written above.

**Comment by Micha-Btz** [comment]
just for clarification we merge all new patches always to the test branch and if we are readyy you will merge it to the cm-12.1 branches for release, right?

**Comment by derfelot** [comment]
yeah, that's the idea. This way we can test build first using those test branches.

I'm getting a funky error during build now (kernel compilation step). I'm building for z3c right now though. I just did a complete cleaning (including ccache) and trying again.

**Comment by derfelot** [comment]
right, that error was because of something strange on my side it seem. a clean build has worked fine.

**Comment by MSe1969** [comment]
I think we should close this issue and only continue in the April and May threads.
Would you agree?

**Comment by Micha-Btz** [comment]
I would leave this open for general questions and create an monthly issue for the better overview.

**Comment by MSe1969** [comment]
Hi @derfelot @Micha-Btz 

please have a look here - you are listed as contributors, so you should be able to edit the OP as well.

https://forum.xda-developers.com/sony-xperia-z1-compact/orig-development/lineageos-12-1-z1c-current-sec-patches-t3614936

**Comment by Viktor245** [comment]
Hi

Good job, I'll write your mistakes here.

**Comment by Viktor245** [comment]
https://play.google.com/store/apps/details?id=net.sourceforge.opencamera

I advise the camera to use this

**Comment by Viktor245** [comment]
http://4pda.ru/forum/index.php?showtopic=446121

Browser recommend this

**Comment by Viktor245** [comment]
http://4pda.ru/forum/index.php?showtopic=589149

The camera can be taken here, vklyuchaya fashion

**Comment by Viktor245** [comment]
I advise you to use this gallery:
http://4pda.ru/forum/index.php?showtopic=785182

**Comment by MSe1969** [comment]
Привет, Виктор - Спасибо за ваши материалы
(That's from Google translate - as far as I know, nobody of us is able to understand Russian :) )

Our stand point is: Everybody can install these apps on his own, so there is no need for us to ship them. We only replace or remove apps in the ROM, if the original app is faulty or insecure and should not be used any more.

The 'open camera' app however could be worth looking at... (but still, everybody can install on his own)
@Micha-Btz @hahnjo @derfelot - any thoughts?

**Comment by Viktor245** [comment]
I agree with you, and I do so, remove unnecessary programs from the system and install my own ones without a binding to google

**Comment by Micha-Btz** [comment]
@Viktor245 you can install the most apps you mentioned from f-droid store so it makes no sense to include this. And as @MSe1969 say we must reduce the stuff we should maintain on our own. If you don't want some app you can take a look at my removal scripts and custom it to your needs.
https://github.com/Micha-Btz/freecyngn

**Comment by Viktor245** [comment]
Hello everybody!

It is easier for me to remove everything in the system and correct the translation.
And so good rum I wish you good luck!

**Comment by MSe1969** [comment]
Hi all,

I have now merged the cm-12.1-test branches into cm-12.1, updated the local manifest in cm-12.1 and also in cm-12.1-test and deleted the test branches (exceptions listed below).

This is the new status:
1. Below repositories have been merged into cm-12.1 and the test branches have been deleted; bot local manifests point to cm-12.1 (note: the cm-12.1-test local manifest always points to cm-12.1-test for build):
build
android_frameworks_av
android_frameworks_base
frameworks/native
external/tremolo
external/libhevc
sonivox

2. I have left the **kernel** repo untouched. 'cm-12.1' is identical to 'cm-12.1-test', but this is because cm-12.1-test contains reversed commits. The local repo for cm-12.1-test still points to cm-12.1-test for our kernel, because we have still our open issue for CVE-2016-5349

3. **bionic** (one backport) has been merged, so cm-12.1 and cm-12.1-test are identical, but I have left the local manifest in cm-12.1-test point to cm-12.1-test, because of our open issue of CVE-2017-0670

4. Same as above (bionic) applies for **boringssl**, because I will open a new issue for that one

I'll update now the XDA thread



**bionic
**boringssl

**Comment by MSe1969** [comment]
With regards to kernel, I have yesterday wrongly looked at the stuff (it DOES make a difference, if you compare branches at github, which branch is the base - and the message if cm-12.1-test is the base and cm-12.1 is the compare branch looks similar to the real "identical" message).
So indeed, cm-12.1-test still needs to be merged into cm-12.1 - will do a test build before and then do the merge.

**Comment by MSe1969** [comment]
I have created a text file in the discussion repo with the XDA formatted install instructions, in case they get lost again ;)

**Comment by MSe1969** [comment]
@Micha-Btz @hahnjo - I've just gone through my repos used for this project and added you as collaborators - no action required yet, but better to have that access granted

**Comment by hahnjo** [comment]
It would probably be easier to have all repositories in an organization...

**Comment by MSe1969** [comment]
I agree. Any preference for name? 
I spontaneously suggest 'amami-lineage' or 'amami-dhmm' (the meaning of the 1st one is obvious, the 2nd would simply be the 1st letters of our github nicknames in alphabetical oder). Other proposals welcome as well. I'll set it up, then, if you agree (unless one of you wishes to set it up). 

We should still do the September ASB "as is" and work in parallel on the org.
We should then setup our complete own manifest in a repo instead of having an ever-growing local manifest and a slight mod in 'build' to name the product output accordingly.

Thoughts? Maybe discuss in an own issue?

**Comment by hahnjo** [comment]
`cm-12.1-amami` or the other way round because we really don't provide a current LineageOS (14.1 or 15).

Agree on your other points, maybe having an own issue would be good.

**Comment by MSe1969** [comment]
#20

**Comment by MSe1969** [comment]
As we will continue in organization "cm12-amami", it's time to close this issue.
